designs/mixins.md

```diff
@@ -692,12 +698,88 @@ The members are ordered as follows:
 - `sizeFilter`
 
 
+### Mixins on shapes with non-member properties
+
+Some shapes don't have members, but do have other properties. Adding a mixin
+to such a shape causes the properties of the other shape to be merged into
+the shape. Scalar properties defined in the local shape are kept, and
+non-scalar properties are merged. When merging map properties, the values for
+local keys are kept. The ordering of merged lists / sets follows the
+same ordering as members.
+
+For example, in the following model:
+
+```
+operation OperationA {}
+
+service A {
+    version: "A"
+    operations: [OperationA]
+}
+
+operation OperationB {}
+
+service B with A {
```

Comment:
Doesn’t this need a mixin trait? Any shape used as a mixin previously needed a mixin trait.